cfme/ansible_tower/jobs.py

```diff
@@ -51,14 +60,80 @@ def is_displayed(self):
         )
 
 
+class AnsibleTowerJobsDetailsView(TowerJobsView):
+    @View.nested
+    class entities(View):  # noqa
+        """ Represents details page when it's switched to Summary/Table view """
+        properties = SummaryTable(title="Properties")
+        relationships = SummaryTable(title="Relationships")
+        smart_management = SummaryTable(title="Smart Management")
+
+    @property
+    def is_displayed(self):
+        """Is this view being displayed?"""
+        return self.title.text == '{} (Summary)'.format(self.context['object'].template_name)
+
+
 @attr.s
-class TowerJobs(BaseEntity):
-    pass
+class TowerJob(BaseEntity):
+    template_name = attr.ib()
+
+    @property
+    def status(self):
+        view = navigate_to(self, 'Details')
+        return view.entities.properties.get_text_of("Status")
+
+    def delete(self):
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Remove this Job', handle_alert=True)
+        view.flash.assert_no_error()
+
+    def is_job_successful(self):
+        return True if self.status == "successful" else False
+
+    def wait_for_completion(self, num_sec=1200, delay=10):
+        view = navigate_to(self, 'Details')
+
+        def last_status():
+            logger.info("Last status message in UI: '{}'".format(
+                view.entities.properties.get_text_of("Status")))
+
+        wait_for(self.is_job_successful, num_sec=num_sec, delay=delay, fail_func=last_status,
```

Comment:
```suggestion
        wait_for(self.is_job_successful, num_sec=num_sec, delay=delay, fail_func=self.last_status,
```

Reply:
This line works fine as is since last_status() is defined within wait_for_completion().

Reply:
Oh whoops, I believe I meant `self.status` rather than `self.last_status`

Reply:
I'd like to use `last_status` as the fail_func just because it's logging this useful  info.
logger.info("Last status message in UI: '{}'".format(self.status))

So, I'm retaining the original statement.